python/ray/util/client/server/proxier.py

```diff
@@ -96,14 +111,25 @@ def _get_unused_port(self) -> int:
                 return port
         raise RuntimeError("Unable to succeed in selecting a random port.")
 
+    def _get_redis_address(self) -> str:
+        """
+        Returns the provided Ray Redis address, or creates a new session.
+        """
+        if self.redis_address:
+            return self.redis_address
+        connection_tuple = ray.init(address=self.redis_address)
```

Comment:
should this start a ray driver?
If yes, why do we need to pass self.redis_address to address? if the if statement did not happen, doesn't it mean that the value here is None?

Reply:
Ah yes, this will be `None`. The logic is the same, but it is quite weird!